### DISCUSSION TITLE
Register and forgot password functionalities

### DISCUSSION THREAD

**Discussion by foxbille** [codeigniter4/shield#500] [Q&A]
Hello,
Can someone explain how to customize Email.fromEmail and Email.fromName which are needed by "register" and "forgot password" functions...
And may this explanation cover all the process of what you have to do to manage things when a user registers or need to recover his password. All of this stuff was not include in the doc... or maybe am i dumb...
Thanks a lot
Eric

**Comment by kenjis** [comment]
Set Config\Email.
https://github.com/codeigniter4/CodeIgniter4/blob/9e479561494c490f5e872569f0202dc1a668b79a/app/Config/Email.php#L12

**Reply by foxbille** [reply]
Ok, that's for the first question.
I have tried, it works. And as i was worried to modify Config\email, i rather put it config in .env and obviously, it works as well.
As i'm using shield and discovering "setting" library, i will also try to store them in "settings" table instead.
Thanks for your help

**Reply by kenjis** [reply]
You need to call `set()` method to save values in the database.
See https://github.com/codeigniter4/settings#usage

**Reply by kodetumbuh** [reply]
> Ok, that's for the first question. I have tried, it works. And as i was worried to modify Config\email, i rather put it config in .env and obviously, it works as well. As i'm using shield and discovering "setting" library, i will also try to store them in "settings" table instead. Thanks for your help

could you explain more step by step because in doc is not included how to setting email, unfortunately this package is really good but doesn't have proper doc

**Reply by kenjis** [reply]
See https://github.com/codeigniter4/shield/issues/289#issuecomment-1178716611

**Comment by datamweb** [comment]
Hi,
please ask the questions separately, you have raised several questions in one topic, this destroys efficiency, both for answering and for the use of other users.

> need to recover his password

There are two solutions for this.

**The first method:** use sessions

modify the `App\Config\Auth.php`
```php
public function loginRedirect(): string
{
    if (session('magicLogin')) {
       return redirect()->route('set_password');
    }

    $url = setting('Auth.redirects')['login'];

    return $this->getUrl($url);
}
```
`set_password` is a method in your custom controller.
Change the password in this process.

```php
$users = model('UserModel');
$user  = $users->findById(auth()->user_ir);

$user->fill([
    'password' => 'New passs'
]);
$users->save($user);
```

**The second method:** using events
https://github.com/codeigniter4/shield/blob/develop/docs/events.md#magiclogin
